internal/handlers/pg/pgdb/database_metadata.go

```diff
@@ -218,20 +225,24 @@ func (m *metadata) remove(ctx context.Context) error {
 	return lazyerrors.Error(err)
 }
 
-// formatCollectionName returns collection name in form <shortened_name>_<name_hash>.
-// Changing this logic will break compatibility with existing databases.
-func formatCollectionName(name string) string {
+// transform returns name in form <shortened_name>_<name_hash>.
+// It replaces special characters with `_`.
+//
+// Deprecated: this function usage is allowed for collection metadata creation only.
+func transform(name string) string {
 	hash32 := fnv.New32a()
 	must.NotFail(hash32.Write([]byte(name)))
 
+	transformed := specialCharacters.ReplaceAllString(name, "_")
+
 	nameSymbolsLeft := maxTableNameLength - hash32.Size()*2 - 1
-	truncateTo := len(name)
+	truncateTo := len(transformed)
 
 	if truncateTo > nameSymbolsLeft {
 		truncateTo = nameSymbolsLeft
 	}
 
-	return name[:truncateTo] + "_" + fmt.Sprintf("%x", hash32.Sum([]byte{}))
+	return transformed[:truncateTo] + "_" + fmt.Sprintf("%x", hash32.Sum([]byte{}))
```

Comment:
```suggestion
	return transformed[:truncateTo] + "_" + fmt.Sprintf("%x", hash32.Sum(nil))
```

That's more idiomatic.

There is also a subtle problem with that line: `%x` omits leading zeros, and that could cause, in extreme cases, clashes. The valid syntax is `%08x`

Reply:
Let me update this to the valid syntax :+1: